pyzx/rewrite.py

```diff
@@ -0,0 +1,143 @@
+# PyZX - Python library for quantum circuit rewriting
+#        and optimization using the ZX-calculus
+# Copyright (C) 2018 - Aleks Kissinger and John van de Wetering
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the ZX-diagram rewrite class of PyZX.
+It is part of a major reworking of how the rule and simplify files 
+work to perform the rewrite rules on diagrams."""
+
+from typing import List, Callable, Optional, Union, Generic, Set, Tuple, Dict, Iterator, cast
+from .graph.base import BaseGraph, VT, ET
+
+class Rewrite(Generic[VT, ET]):
+
+    def __init__(self) -> None:
+        pass
+
+    def simp(self, graph) -> None:
```

Comment:
Feedback based on discussion with Aleks: The simp and apply methods should return a bool te denote whether they actually did any rewriting.